pkg/bootstrap/node.go

```diff
@@ -506,3 +509,194 @@ func ensureConntrackHashsize() {
 		return
 	}
 }
+
+func setIptablesForwardPolicy() error {
+	var cmdPath string
+	var err error
+
+	log.Println("making iptables FORWARD chain defaults to ACCEPT...")
+
+	if cmdPath, err = exec.LookPath("iptables"); err != nil {
+		// fall back to an ordinary abspath
+		cmdPath = "/sbin/iptables"
+	}
+
+	// set the default policy for FORWARD chain to ACCEPT
+	// : iptables -P FORWARD ACCEPT
+	args := []string{
+		cmdPath,
+		"-P",
+		"FORWARD",
+		"ACCEPT",
+	}
+
+	cmd := exec.Cmd{
+		Path:   cmdPath,
+		Args:   args,
+		Env:    os.Environ(),
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isCniRuleLoaded() bool {
+	var cmdPath string
+	var err error
+
+	if cmdPath, err = exec.LookPath("iptables"); err != nil {
+		// fall back to an ordinary abspath
+		cmdPath = "/sbin/iptables"
+	}
+
+	// check if a cni iptables rules already exists
+	// : iptables -C FORWARD -i cni0 -j ACCEPT
+	args := []string{
+		cmdPath,
+		"-C",
+		"FORWARD",
+		"-i",
+		"cni0",
+		"-j",
+		"ACCEPT",
+	}
+
+	cmd := exec.Cmd{
+		Path:   cmdPath,
+		Args:   args,
+		Env:    os.Environ(),
+		Stdout: os.Stdout,
+	}
+
+	if err := cmd.Run(); err != nil {
+		// error means that the rule does not exist
+		return false
+	}
+
+	return true
+}
+
+func setAllowCniRule() error {
+	var cmdPath string
+	var err error
+
+	if cmdPath, err = exec.LookPath("iptables"); err != nil {
+		// fall back to an ordinary abspath
+		cmdPath = "/sbin/iptables"
+	}
+
+	// insert an iptables rules to allow traffic through cni0
+	// : iptables -I FORWARD 1 -i cni0 -j ACCEPT
+	args := []string{
+		cmdPath,
+		"-I",
+		"FORWARD",
+		"1",
+		"-i",
+		"cni0",
+		"-j",
+		"ACCEPT",
+	}
+
+	cmd := exec.Cmd{
+		Path:   cmdPath,
+		Args:   args,
+		Env:    os.Environ(),
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ensureIptables() {
+	setIptablesForwardPolicy()
+
+	if !isCniRuleLoaded() {
+		log.Println("setting iptables rule to allow CNI traffic...")
+		if err := setAllowCniRule(); err != nil {
+			log.Printf("error running iptables: %v\n", err)
+			return
+		}
+	}
+}
+
+func isSELinuxEnforcing() bool {
+	var cmdGetPath string
+	var err error
+
+	if cmdGetPath, err = exec.LookPath("getenforce"); err != nil {
+		// fall back to an ordinary abspath
+		cmdGetPath = "/usr/sbin/getenforce"
+	}
+
+	argsGet := []string{
+		cmdGetPath,
+	}
+
+	cmdGet := exec.Cmd{
+		Path:   cmdGetPath,
+		Args:   argsGet,
+		Env:    os.Environ(),
+		Stderr: os.Stderr,
+	}
+
+	// As getenforce always returns non-error, we should ignore the error.
+	// Instead, parse the output string directly to determine the current
+	// SELinux mode.
+	outstr, _ := cmdGet.Output()
+	sestatus := strings.TrimSpace(string(outstr))
+	if sestatus == "Enforcing" {
+		return true
+	}
+
+	return false
+}
+
+func disableSELinuxEnforcing() error {
```

Comment:
I thought we said we didn't want to do this without user intervention. Maybe @iaguis recalls.